infra/macos/2-common-box/init.sh

```diff
@@ -114,10 +114,14 @@ echo "created cache partitions"
 
 # Note: installing Nix in single-user mode with /nix already existing and
 # writeable does not require sudoer access
+# Single-user install is no longer available in the 2.4+ installer, so we pin
+# _the installer_ to 2.3.16 then upgrade to nix 2.4.
 su -l vsts <<'END'
 set -euo pipefail
 export PATH="/usr/local/bin:/usr/sbin:$PATH"
-bash <(curl -sSfL https://nixos.org/nix/install)
+bash <(curl -sSfL https://releases.nixos.org/nix/nix-2.3.16/install)
+export PATH="$HOME/.nix-profile/bin:$PATH"
+nix-channel --update && nix-env -iA nixpkgs.nix_2_4
```

Comment:
This is non-deterministic, you’ll upgrade to newer patch releases potentially. I _believe_ the installers directly do not suffer from that problem. I’m not hugely worried about this but it’s probably at least worth keeping in mind and maybe worth documenting in a comment.

Reply:
In my mind, that was implied in the `2.4` above (since `2.4` on its own is not a version), but I've made that more explicit by saying `2.4.x` now.

(It's also not worse than before, mind.)